applications/blockpublisher.md

```diff
@@ -0,0 +1,235 @@
+
+### Milestone 2:
+
+- Leptos UI Development:
+
+Design and develop a user-friendly UI for article upload, viewing, and interaction with the Substrate templates using Leptos.
+Integrate the Leptos UI with the Substrate templates to enable a seamless user experience.
+
+- **Estimated duration:** 6 month
+- **FTE:**  1,3
+- **Costs:** 10,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License |  GPL |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user about the project |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an **article**/workshop that explains our project (what was done/achieved as part of the grant.) |
+| 1. | Leptos UI Development | Design and develop a user-friendly UI for article upload, viewing, and interaction with the Substrate templates using Leptos.|
+
+
+### Milestone 3:
+
+- Zero-Knowledge Proof (ZKP) Integration:
+
+Develop a ZKP system for secure article purchasing, ensuring that buyers can prove ownership of the article without revealing the password.
+Integrate the ZKP system with the Substrate templates and Leptos UI to enable secure and private article purchasing.
+
+- **Estimated duration:** 6 month
+- **FTE:**  1,3
+- **Costs:** 10,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License |  GPL |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user about the project |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an **article**/workshop that explains our project (what was done/achieved as part of the grant.) |
+| 1. | ZKP for article purchase | Develop a ZKP system for secure article purchasing, ensuring that buyers can prove ownership of the article without revealing the password. RiscZero framework is used for ZKP system.|
+| 2. | UI | Integrate the ZKP system with the Substrate templates and Leptos UI to enable secure and private article purchasing. |
+
+
+## Future Plans
+
+- how you intend to finance the project's long-term maintenance and development
+
+Once the token value of the app increases, future maintenance cost and development is done through native treasury or governance of the app.
```

Comment:
What happens on the occasion that the token value doesn't increase? :) Also, what are your plans for the token sale? Please note our [guidelines](https://github.com/w3f/grants-program?tab=readme-ov-file#guidelines) when it comes to on-going token sales.

Reply:
A minimal token value is required for maintenance, even if it doesn't increase. In worst case with 1% inflation, $1 million trading volume per year can pay a full time developer.  It depends on some marketing work.  We don't have on-going token sales. Token sale will only be don't after the project is ready and tested in testnet.